include/flamegpu/util/dstring.h

```diff
@@ -0,0 +1,33 @@
+#ifndef INCLUDE_FLAMEGPU_UTIL_DSTRING_H_
+#define INCLUDE_FLAMEGPU_UTIL_DSTRING_H_
+
+#include <cuda_runtime.h>
+
+namespace flamegpu {
+namespace util {
+/**
+ * Device implementations of required string.h functionality
+ */
+
+
+/**
+ * strcmp() - Compare two strings, return 0 if equal, otherwise return suggests order
+ *
+ * @param s1 First string to be compared
+ * @param s2 Second string to be compared
+ *
+ * @note Implementation based on https://stackoverflow.com/a/34873763/1646387
+ */
+__device__ __forceinline__ int dstrcmp(const char *s1, const char *s2) {
```

Comment:
This will not be resolved at compile time for RTC due to the use of the while loop. If we really wanted to have this resolve at compile time we would have to find a way for this to work with `constexpr` or a through template recursion. I don't think this is necessary though and we should flag that string comparisons are not particularly fast in the docs.